### PR TITLE
Switch TR-55 Water Quality to Bullet Charts

### PR DESCRIPTION
**Pull request by rajadain**
## Overview

Previously the water quality data was represented as a horizontal bar chart, with one bar for each type of observation. Unfortunately, the values of the three observations was quite disparate in scale, making the smallest bar insignificantly small.

To help bring each value to first class citizenship, we switch to using three bullet charts. Each chart is scaled to suit the value thereof, and thus shows real, actionable information.

Connects #2165 

### Demo

image

Tagging @jfrankl and @ajrobbins for visual review.

### Notes

The chart labels in the mock-up are left-aligned, but I couldn't figure out how to left-align the text easily here. Is the right alignment (default design for bullet charts) good enough?

The actual value isn't shown on the chart, but in the tooltip, which looks like this:

image

But enabling that tooltip also enables this one:

image

and I couldn't figure out a way to have one and not the other.

## Testing Instructions

 * Check out this branch and `bundle`
 * Go to [:8000/](http://localhost:8000/) and create or open a TR-55 project
 * Go to the Water Quality tab. Ensure the new chart is there, and the values correspond to the table.

### CODE REVIEW

**Comment by ajrobbins** [comment]
Looks good to me visually! How are we getting the max values? It is a little weird to have that tooltip, but doesn't bother me too much.

**Comment by rajadain** [comment]
Adding a GIF of the tooltip to better communicate:

**Comment by kellyi** [comment]
Taking a look at this!

**Comment by kellyi** [comment]
This looks great!

screen shot 2017-11-09 at 5 20 40 pm

**Comment by kellyi** [comment]
> How are we getting the max values?

Just to clarify: we are calculating the max value as the top of the current value's order of magnitude here?

https://github.com/WikiWatershed/model-my-watershed/pull/2503/files#diff-3e33aa79d87dd300905a2a0b6c714e56R611

That seems reasonable to me: it'd ensure the charts always show an appropriate ratio.

**Comment by rajadain** [comment]
> we are calculating the max value as the top of the current value's order of magnitude 

Yes.

**Comment by kellyi** [comment]
I'm in favor of leaving the chart title & subtitle left justified as is.

I was able to move them around manually in Chrome by adding an attribute like `dx="-3em"` to the `nv-title` and `nv-subtitle` classes, but I don't know whether nvd3 has an interface to do this automatically:

screen shot 2017-11-09 at 5 34 47 pm

screen shot 2017-11-09 at 5 35 49 pm

However, it'd be a chore to have to set those manually & verify correctness across browsers & screen sizes.

File: src/mmw/js/src/core/chart.js [hunk 1]
```diff
@@ -596,9 +596,45 @@ function renderCompareMultibarChart(chartEl, name, label, colors, stacked, yMax,
...
+
+    // Calculate max range of chart
+    // by rounding up to the next place
+    while (data < 1) {
+        data *= 10;
+        adjustment -= 1;
+    }
+    while (data > 10) {
+        data /= 10;
+        adjustment += 1;
```
Comment (kellyi) [inline src/mmw/js/src/core/chart.js]: `data` looks not to be used again after these adjustments (I checked in the REPL and the `measures: [data]` is passed by value.

Consider making a small utility function which would take a value of whatever magnitude and return its base 10 exponent & adding tests for < 1, > 10, and single digit values.

**Comment by ajrobbins** [comment]
I'm fine with it.

**Comment by jfrankl** [comment]
LGTM!

**Comment by rajadain** [comment]
Rebased on latest `develop`, extracted the range in magnitude logic to a utility function and added some tests (and discovered some failing boundary conditions, so good idea @kellyi!). Will merge when green. Thanks for all the reviews!
